File: app/src/main/java/com/alternativeinfrastructures/noise/models/signal/NoiseSignedPreKeyStore.java
package com.alternativeinfrastructures.noise.models.signal;

import org.whispersystems.libsignal.InvalidKeyIdException;
import org.whispersystems.libsignal.state.SignedPreKeyRecord;
import org.whispersystems.libsignal.state.SignedPreKeyStore;

import java.util.List;

public class NoiseSignedPreKeyStore implements SignedPreKeyStore {
    @Override
    public SignedPreKeyRecord loadSignedPreKey(int signedPreKeyId) throws InvalidKeyIdException {
        // TODO
        return null;
    }

    @Override
    public List<SignedPreKeyRecord> loadSignedPreKeys() {
        // TODO
        return null;
    }

    @Override
    public void storeSignedPreKey(int signedPreKeyId, SignedPreKeyRecord record) {
        // TODO
    }

    @Override
    public boolean containsSignedPreKey(int signedPreKeyId) {
        // TODO
        return false;
    }

    @Override
    public void removeSignedPreKey(int signedPreKeyId) {
        // TODO
    }
}

File: app/src/main/java/com/alternativeinfrastructures/noise/models/signal/NoiseSignedPreKeyStore.kt
package com.alternativeinfrastructures.noise.models.signal

import org.whispersystems.libsignal.InvalidKeyIdException
import org.whispersystems.libsignal.state.SignedPreKeyRecord
import org.whispersystems.libsignal.state.SignedPreKeyStore

class NoiseSignedPreKeyStore : SignedPreKeyStore {
    @Throws(InvalidKeyIdException::class)
    override fun loadSignedPreKey(signedPreKeyId: Int): SignedPreKeyRecord? {
        // TODO
        return null
    }

    override fun loadSignedPreKeys(): List<SignedPreKeyRecord>? {
        // TODO
        return null
    }

    override fun storeSignedPreKey(signedPreKeyId: Int, record: SignedPreKeyRecord) {
        // TODO
    }

    override fun containsSignedPreKey(signedPreKeyId: Int): Boolean {
        // TODO
        return false
    }

    override fun removeSignedPreKey(signedPreKeyId: Int) {
        // TODO
    }
}
